pinot-query-runtime/src/main/java/org/apache/pinot/query/service/QueryDispatcher.java

```diff
@@ -118,6 +119,10 @@ public static List<DataTable> reduceMailboxReceive(MailboxReceiveOperator mailbo
     while (true) {
       transferableBlock = mailboxReceiveOperator.nextBlock();
       if (TransferableBlockUtils.isEndOfStream(transferableBlock)) {
+        if (transferableBlock.isErrorBlock()) {
```

Comment:
This does not look correct. The if condition at line 122 is inside the if condition at line 121. How is it possible that a TransferableBlock is both a EOS block and a error block ? Both are metadata block types but we still need to distinguish between them ?

Reply:
correct. end-of-stream means no more blocks will be send. 
it could mean 
- all data has been transferred, and not error
- all data transfer has been halted because of error